Source/WTF/wtf/Vector.h

```diff
@@ -108,217 +94,113 @@ struct VectorInitializer<true, false, T>
-        memset(dst, val, dstEnd - dst);
-    }
-};
-
-template<bool canCompareWithMemcmp, typename T>
-struct VectorComparer;
-
-template<typename T>
-struct VectorComparer<false, T>
-{
-    static bool compare(const T* a, const T* b, size_t size)
-    {
-        for (size_t i = 0; i < size; ++i)
-            if (!(a[i] == b[i]))
-                return false;
-        return true;
-    }
-};
-
-template<typename T>
-struct VectorComparer<true, T>
-{
-    static bool compare(const T* a, const T* b, size_t size)
-    {
-        return memcmp(a, b, sizeof(T) * size) == 0;
-    }
-};
-
-template<typename T>
-struct VectorTypeOperations
-{
-    static void destruct(T* begin, T* end)
-    {
-        VectorDestructor<!std::is_trivially_destructible<T>::value, T>::destruct(begin, end);
-    }
-
-    static void destruct(std::span<T> span)
-    {
-        VectorDestructor<!std::is_trivially_destructible<T>::value, T>::destruct(std::to_address(span.begin()), std::to_address(span.end()));
-    }
-
-    static void initializeIfNonPOD(T* begin, T* end)
-    {
-        VectorInitializer<VectorTraits<T>::needsInitialization, VectorTraits<T>::canInitializeWithMemset, T>::initializeIfNonPOD(begin, end);
-    }
-
-    static void initialize(T* begin, T* end)
-    {
-        VectorInitializer<VectorTraits<T>::needsInitialization, VectorTraits<T>::canInitializeWithMemset, T>::initialize(begin, end);
-    }
-
-    template<typename ... Args>
-    static void initializeWithArgs(T* begin, T* end, Args&&... args)
-    {
-        VectorInitializer<VectorTraits<T>::needsInitialization, VectorTraits<T>::canInitializeWithMemset, T>::initializeWithArgs(begin, end, std::forward<Args>(args)...);
-    }
-
-    static void move(T* src, T* srcEnd, T* dst)
-    {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::move(src, srcEnd, dst);
-    }
 
     static void move(std::span<T> src, std::span<T> dst)
     {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::move(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
+        move(std::to_address(src.begin()), std::to_address(src.end()), std::to_address(dst.begin()));
     }
 
     static void moveOverlapping(T* src, T* srcEnd, T* dst)
     {
-        VectorMover<VectorTraits<T>::canMoveWithMemcpy, T>::moveOverlapping(src, srcEnd, dst);
+        if constexpr (VectorTraits<T>::canMoveWithMemcpy)
+            memmove(static_cast<void*>(dst), static_cast<void*>(const_cast<T*>(src)), reinterpret_cast<const char*>(srcEnd) - reinterpret_cast<const char*>(src));
+        else {
+            if (src > dst)
+                move(src, srcEnd, dst);
+            else {
+                T* dstEnd = dst + (srcEnd - src);
+                while (src != srcEnd) {
+                    --srcEnd;
+                    --dstEnd;
+                    new (NotNull, dstEnd) T(WTFMove(*srcEnd));
+                    srcEnd->~T();
+                }
+            }
```

Comment:
I think we can use `std::uninitialized_move` or `std::ranges_uninitialized_move` here instead of writing our own loop, although we will need to verify that it's still equally efficient.